x-pack/plugins/security_solution/server/lib/risk_engine/utils/retry_es.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import { errors as EsErrors } from '@elastic/elasticsearch';
+
+const MAX_ATTEMPTS = 3;
+
+const retryResponseStatuses = [
+  503, // ServiceUnavailable
+  408, // RequestTimeout
+  410, // Gone
+];
+
+const isRetryableError = (e: Error) =>
+  e instanceof EsErrors.NoLivingConnectionsError ||
+  e instanceof EsErrors.ConnectionError ||
+  e instanceof EsErrors.TimeoutError ||
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  (e instanceof EsErrors.ResponseError && retryResponseStatuses.includes(e?.statusCode!));
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const retryTransientEsErrors = async <T>(
```

Comment:
have you considered using `p-retry` utility, already available in Kibana repo.
It has rich configuration options within it, for example: https://github.com/elastic/kibana/blob/8.8/src/plugins/content_management/server/event_stream/es/init/es_event_stream_initializer.ts#L48-L60

Reply:
This [retry_es.ts](https://github.com/elastic/kibana/pull/158422/files/c9183866e5babb824d71c1d1dd3cfa9d6468d9df#diff-4b7ba1a0f31c3bbea878f941f2fbce056c8b0ce05ea4ad6754ab68dde931629a) is copy past from alerting plugin, with the intention just use the creation of DS in the future from this plugin.

Reply:
I would rename this file to be more accurate/similar to its origin: `retry_transient_es_errors.ts`